data-jdbc/src/main/java/io/micronaut/data/jdbc/operations/DefaultJdbcRepositoryOperations.java

```diff
@@ -324,11 +324,14 @@ public <T, R> R findOne(@NonNull PreparedQuery<T, R> pq) {
                         return result;
                     } else if (rs.next()) {
                         if (preparedQuery.isDtoProjection()) {
-                            TypeMapper<ResultSet, R> introspectedDataMapper = new DTOMapper<>(
-                                    persistentEntity,
-                                    columnNameResultSetReader,
-                                    jsonCodec,
-                                    conversionService);
+                            boolean isRawQuery = preparedQuery.getAnnotationMetadata().stringValue(Query.class, DataMethod.META_MEMBER_RAW_QUERY).isPresent();
+                            TypeMapper<ResultSet, R> introspectedDataMapper = new SqlDTOMapper<>(
+                                persistentEntity,
+                                isRawQuery ? getEntity(preparedQuery.getResultType()) : persistentEntity,
```

Comment:
Need to pass DTO entity so we can take persistent property if not found in mapped entity. This is how findAll works and findOne should work the same way.